### PR TITLE
PerformanceObserverEntryList should not extend Array

### PR DESCRIPTION
**Pull request by phamann**
*Note:* This is a PR into #17 until hat is merged. 

### TL;DR
Fixes a hidden bug with current implementation of the library which had the `EntryList` implementation extend the native `Array` and incorrectly suggest that it implements the `PerformanceEntryList` type and not the correct `PerformanceObserverEntryList` interface.

It seems this bug was obscured from the compiled output via Webpack/babel, which seems to treat compilation of `Class foo extends Array` (and all its inherent problems) differently that theTypeScript compiler, and was exposed when we removed those tools in #17 and broke upstream test suites.

Fortunately, per the spec, [`PerformanceObserverEntryList`](https://w3c.github.io/performance-timeline/#performanceobserverentrylist-interface) doesn't need to be an array like object and just needs to implement the interface, which we already were. This PR fixes both of these issues.

### CODE REVIEW

**Review by doramatadora — APPROVED** [review]
This is. So. Nice 👏🏼  Using polymorphism over inheritance to fix a pretty obscure bug.
